api/enki.py

```diff
@@ -37,41 +41,52 @@
 
 from .circulation import BaseCirculationAPI, FulfillmentInfo, LoanInfo
 from .circulation_exceptions import *
-from .selftest import HasSelfTests, SelfTestResult
-
+from .selftest import HasCollectionSelfTests, SelfTestResult
 
-class EnkiAPI(BaseCirculationAPI, HasSelfTests):
 
+class EnkiConstants:
     PRODUCTION_BASE_URL = "https://enkilibrary.org/API/"
 
-    ENKI_LIBRARY_ID_KEY = "enki_library_id"
-    DESCRIPTION = _("Integrate an Enki collection.")
-    SETTINGS = [
-        {
-            "key": ExternalIntegration.URL,
-            "label": _("URL"),
-            "default": PRODUCTION_BASE_URL,
-            "required": True,
-            "format": "url",
-        },
-    ] + BaseCirculationAPI.SETTINGS
-
-    LIBRARY_SETTINGS = BaseCirculationAPI.LIBRARY_SETTINGS + [
-        {"key": ENKI_LIBRARY_ID_KEY, "label": _("Library ID"), "required": True},
-        {
-            "key": ExternalIntegration.DISPLAY_RESERVES,
-            "label": _("Show/Hide Titles with No Available Loans"),
-            "required": False,
-            "description": _(
+
+class EnkiSettings(BaseSettings):
+    url: Optional[str] = FormField(
```

Comment:
Does this need to be optional? And can we do validation with the `HttpUrl` type rather then adding `format="url"` below?